From e00329729e1fd18fa452e91820a81964d5527e6d Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@collabora.com>
Date: Tue, 28 Apr 2020 15:04:19 +0200
Subject: [PATCH] sw from-bottom relative orientation: add UNO API

When it comes to vertical positioning of anchored sw objects, one can
say the position should be "1cm from the top of the page". But measuring
from the bottom of something was not possible.

Add API for this to help working with documents from Word, which
supports the feature.

There is no duplicated C++ enum in sw/ for vertical relative
orientation, so no "doc model" changes are needed for this in sw/.

Change-Id: I3199d3e794bda2f21f92ce3bb7c3c6f04d284db2
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/93065
Reviewed-by: Miklos Vajna <vmiklos@collabora.com>
Tested-by: Jenkins
(cherry picked from commit 79107d3f8d10aa0f38641775c5eb47dcfd4fd37e)
---
 offapi/com/sun/star/text/RelOrientation.idl | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/offapi/com/sun/star/text/RelOrientation.idl b/offapi/com/sun/star/text/RelOrientation.idl
index adfb094fd593..59f39a045740 100644
--- a/offapi/com/sun/star/text/RelOrientation.idl
+++ b/offapi/com/sun/star/text/RelOrientation.idl
@@ -73,6 +73,12 @@ published constants RelOrientation
          */
         const short TEXT_LINE = 9;
 
+    /** Similar to PAGE_PRINT_AREA, but count from bottom, not from top.
+
+        @since LibreOffice 7.0
+     */
+    const short PAGE_PRINT_AREA_BOTTOM = 10;
+
 };
 
 
-- 
2.26.1

